geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -935,7 +935,13 @@ public static GemFireCacheImpl getForPdx(String reason) {
       }
     } // synchronized
 
-    clientMetadataService = new ClientMetadataService(this);
+    if (this.isClient) {
```

Comment:
Instead of adding this new block of code look for existing code that has "Running in client mode" and sets the resourceEventsListener to null. It is around line 831. In that else block call the new ClientMetaDataService and before the else set it to null.